website/src/components/NavBar.jsx

```diff
@@ -32,49 +34,86 @@ const styles = {
 };
 
 /**
- * Gives HTML for the NavBar.
- * @returns {HTML}   The HTML of the navigation bar.
+ * @classdesc The nav bar for the page.
  */
-function NavBar() {
-  return (
-    <AppBar
-      title="CoMake"
-      iconElementRight={
-        <div>
-          <Link to="/profile">
-            <FlatButton
-              secondary={true}
-              style={styles.btn}
-            >
-              <ListItem
-                disabled={true}
-                leftAvatar={
-                  <Avatar src="http://placekitten.com/95/95" />
-                }
-                style={styles.user}
+class NavBar extends React.Component {
+  /**
+   * Constructor for the class.
+   * @param {Object} props The props to be passed in.
+   */
+  constructor(props) {
+    super(props);
+  }
+
+  /**
+   * Function to be triggered on NavBar mounting, fetches user's information.
+   * @returns {void}
+   */
+  componentDidMount() {
+    getUserInfo(this.props.dispatch);
+  }
+
+  /**
+   * Renders the NavBar.
+   * @returns {HTML} The html to be dispalyed.
+   */
+  render() {
+    let photoURL = "";
+    let username = "";
+    if (this.props.userInfo) {
+      photoURL = this.props.userInfo.photo;
+      username = this.props.userInfo.name;
+    }
+    return (
+      <AppBar
+        title="CoMake"
+        iconElementRight={
+          <div>
+            <Link to="/profile">
+              <FlatButton
+                secondary={true}
+                style={styles.btn}
               >
-                <span>
-                  Vin Diesel <Down color={white} style={styles.dropdownIcon} />
-                </span>
-              </ListItem>
-            </FlatButton>
+                <ListItem
+                  disabled={true}
+                  leftAvatar={
+                    <Avatar src={photoURL} />
+                  }
+                  style={styles.user}
+                >
+                  <span>
+                    {username}
+                    <Down color={white} style={styles.dropdownIcon} />
+                  </span>
+                </ListItem>
+              </FlatButton>
+            </Link>
+            <FlatButton
+              label={"Log Out"}
+              labelStyle={styles.user}
+              style={[styles.user, styles.btn]}
+              onClick={() => signOut()}
```

Comment:
since signOut is just a function, you can just put onClick={signOut} instead of putting it in an anonymous function